Add containers from Letheia Expanded and bk3k Blocks and Objects

OmegaPaladin opened the pull request:
I created patch files for every container in these two mods (bk3k only has one, but it is huge and needs sorting) using the exact code used in other mod patches previously included in the project

.  Letheia Expanded containers are going to be comparatively small, but I did not want to mess with slot size right now. 

https://steamcommunity.com/sharedfiles/filedetails/?id=757860462
https://steamcommunity.com/sharedfiles/filedetails/?id=1987992979

OmegaPaladin committed:
Create bk3k_stationWarehouseLocker.object.patch

Added the huge locker in bk3k Blocks and Objects
https://steamcommunity.com/sharedfiles/filedetails/?id=757860462

OmegaPaladin committed:
Create letheiaexpandedlegioncontainer1.object.patch

Adding the first container for the Legion

OmegaPaladin committed:
Uploaded other files

Duplicates of the script for the other Letheia containers

OmegaPaladin committed:
Create letheiaexpandedbookcasetall.object.patch

Added Letheia Expanded Bookcase
https://steamcommunity.com/sharedfiles/filedetails/?id=1987992979

OmegaPaladin committed:
Add files via upload

Added the rest of the Letheia Expanded containers.

Kherae commented:
just to make sure, you completely tested this first, right? The lack of metadata changes says "probably not". you'll need to find the "name" field from those two mods' metadata files (not "friendlyName") to add to includes

OmegaPaladin committed:
Update .metadata

Added includes for the two mods

OmegaPaladin commented:
I have updated the includes in the metadata, but this needs more testing.  Like many things, it seemed a lot simpler at first glance.

What exactly is needed with the supportedcontainers.config.patch - do **all** of the items in the patch need to be added to that file?

Kherae commented:
any containers that are to be supported by the storage expander

OmegaPaladin committed:
Update supportedcontainers.config.patch

Need comments on this as I am not sure on where the error is

OmegaPaladin committed:
Delete objects/letheiaobjectsexpanded directory

OmegaPaladin committed:
Rename objects/shipyard/bk3k_stationWarehouseLocker.object.patch to objects/shipyard/bk3k_stationWarehouseLocker/bk3k_stationWarehouseLocker.object.patch

OmegaPaladin committed:
Delete objects/legion directory

OmegaPaladin committed:
Delete objects/letheiaobjectsexpanded directory

OmegaPaladin committed:
Delete objects/shipyard directory

OmegaPaladin committed:
Rebuilt directory structure

Deleting and reuploading was faster than editing each file

OmegaPaladin committed:
Delete objects/legion directory

OmegaPaladin committed:
Rebuilt Directory structure

Did this to match the main build

OmegaPaladin committed:
Update supportedcontainers.config.patch

Damn it, forgot to close parantheses

OmegaPaladin committed:
Update supportedcontainers.config.patch

fixed copy paste error

OmegaPaladin commented:
I was able to get the Letheia Expanded containers working partially - they can be sorted and carried with objects inside.  
The problem with the bk3k container is that it 399 slots, and that is not a supported size with Enhanced storage.  This causes a crash.  I'm still debating whether to make it 300 slots, 540 slots, or just skip it.  (I removed it for the time being) What's listed here will work and has been tested with multiple containers from Letheia Expanded.

If you have suggestions on which approach to take with the bk3k container, let me know.

I created another fork to test out the supported containers patch, since it was causing errors in the lua script for the storage Expander.   If you could take a look at the code in that branch, I'd appreciate it

Kherae commented:
a simple json linter would have answered that. COMMAS. your supported containers patch is missing plenty.

OmegaPaladin committed:
Update supportedcontainers.config.patch

JSONLint is my new best friend.
Tested with both mods and it works

OmegaPaladin committed:
Update bk3k_stationWarehouseLocker.object.patch

Now works, although with a lower storage capacity than before.  I used code from the Enhanced Storage mod here

OmegaPaladin committed:
Merge pull request #1 from OmegaPaladin/OmegaPaladin-patch-1

Jsonlint applied to both changes

OmegaPaladin commented:
Sorry about not getting back to you sooner, work was keeping me busy.  I have applied sufficient commas to the patch files.

I was able to upgrade storage on Letheia Expanded objects with no problems, and the bk3k container is now a 300 slot container.

This was certainly a learning experience, and I appreciate your patience.

Kherae commented:
at least you  didnt completely forget this exists. mind changing something? due to...potential issues...I'd rather you swap the bk3k container to something equal to/higher than the original

OmegaPaladin committed:
Update bk3k_stationWarehouseLocker.object.patch

Changed to a larger value that the default 399.  540 is the next chest size that has a UI.

OmegaPaladin commented:
The next level of container supported by Enhanced Storage has 540 slots.  (there's no 400 or 500 slot option for chest interfaces)  I increased it to that level.

Would you suggest increasing the container sizes for Letheia Expanded to bring them in line with other enhanced storage containers, or just leave them as is?

Kherae commented:
1) are they smaller than the originals? If yes, increase
2) are they causing UI errors? if yes, increase
3) if 1 and 2 fail, no.
